Document logging configuration (#3397)
tarantool/doc: doc/reference/configuration/cfg_logging.rst

.. confval:: log_level

    Since version 1.6.2.

>     Specifies the level of detail the :ref:`log <admin-logs>` has. There are seven levels:

Reviewer: Maybe just drop the verb?
"The level of detail..."

Author: I'd keep `Specifies` here as it shows that the property value can be get and set. What do you think?

    * 6 – ``VERBOSE``
    * 7 – ``DEBUG``

    By setting ``log_level``, you can enable logging of all events with severities above
    or equal to the given level. Tarantool prints logs to the standard

>     error stream by default, but this can be changed with the

Reviewer: ```suggestion
    error stream by default. This can be changed with the
```

    | Dynamic: **yes**

>     Warning: prior to Tarantool 1.7.5 there were only six levels and ``DEBUG`` was

Reviewer: Maybe turn this into a real [warning](https://www.tarantool.io/en/doc/latest/contributing/docs/markup/admonitions/)?

Author: Converted to a note

    Since version 1.7.4.
    By default, Tarantool sends the log to the standard error stream

>     (``stderr``). If ``log`` is specified, Tarantool can send the log to a ...

Reviewer: Not sure that the list below improves the text significantly: all the options are just 1-2 words and don't have explanations.

Reviewer: Anyway, in this case there are punctuation issues:
- `...` instead of `:` before the list
- `;` in the end of list items (see [our style guide](https://www.tarantool.io/en/doc/latest/contributing/docs/style/#ending-punctuation-in-lists-and-tables).

Author: > Not sure that the list below improves the text significantly: all the options are just 1-2 words and don't have explanations.

Generally, I agree but still think it's better than three `or` in a row :)

    * file;

    * pipe;

    * system logger.

>     The example below shows how to send logs to the ``tarantool.log`` file:

Reviewer: Idea: make examples less "narrative" but more structured: 

```suggestion
    Example 1: sending the log to the ``tarantool.log`` file.
```

Author: Yep, I like it.

    .. code-block:: lua

        box.cfg{log = 'tarantool.log'}
        -- or
        box.cfg{log = 'file:tarantool.log'}

    This opens the file ``tarantool.log`` for output on the server's default
    directory. If the ``log`` string has no prefix or has the prefix "file:",
    then the string is interpreted as a file path.

>     This example shows how to send the log to a pipe:

Reviewer: ```suggestion
    Example 2: sending the log to a pipe.
```

    .. code-block:: lua

        box.cfg{log = '| cronolog tarantool.log'}
        -- or
        box.cfg{log = 'pipe: cronolog tarantool.log'}

    This starts the program `cronolog <https://linux.die.net/man/1/cronolog>`_ when the server starts, and
    sends all log messages to the standard input (``stdin``) of ``cronolog``.
    If the ``log`` string begins with '|' or has the prefix "pipe:",
    then the string is interpreted as a Unix
    `pipeline <https://en.wikipedia.org/wiki/Pipeline_%28Unix%29>`_.

>     The example below shows how to send the log to syslog:

Reviewer: ```suggestion
    Example 3: sending the log to syslog.
```

    If the ``log`` string begins with "syslog:", then it is
    interpreted as a message for the
    `syslogd <https://linux.die.net/man/8/syslogd>`_ program, which normally
    is running in the background on any Unix-like platform.

>     The setting can be 'syslog:', 'syslog:facility=...', 'syslog:identity=...',

Reviewer: ```suggestion
    The setting can be ``syslog:``, ``syslog:facility=...``, ``syslog:identity=...``,
```
